pleskdistup/common/src/mariadb.py

```diff
@@ -93,12 +93,12 @@ def get_installed_mariadb_version() -> MariaDBVersion:
 
 
 def get_mariadb_config_file_path() -> str:
-    if dist._is_rhel_based(dist.get_distro()):
+    if dist.get_distro().rhel_based:
```

Comment:
I think it would be more natural to write `dist.get_distro().is_rhel_based()` instead. Perhaps we should consider making it a method rather than a parameter.

Reply:
Technically it's already a method, just in read-only property form. Since this value can't depend on any parameters, making it an ordinary method and adding `is_` and `()` on any invocation seems redundant to me.

Reply:
In my opinion, the issue is more about readability than technicality. I think using 'if distro is rhel based' looks better than 'if distro rhel based' (because of 'is' obviously).
Additionally, I would expect to have a setter for a parameter, or at least it can be naturally added. However, no one will question why I can't add a setter for the `is_rhel_based` method.

Reply:
It looks no worse than check of any boolean variable named like `rhel_based`.
Not having a setter is normal for properties, or why else read-only properties were added to Python in the first place? :)

Reply:
> It looks no worse than check of any boolean variable named like rhel_based.

Yep, it looks worse then checking `is_rhel_based` method =)

> Not having a setter is normal for properties, or why else read-only properties were added to Python in the first place? :)

As far as I know we can't prevent from adding setter for property and can't evidently show it should be read-only. I might be wrong at this.

Reply:
If you want to prevent someone from doing something, Python is not the language you're looking for :)
For example, anyone can add `set_is_rhel_based(x)` later or do `obj.is_rhel_based = foobar` either.
But assignment to a read-only property would result in an error, so you can't inadvertently overwrite it.

Reply:
Some insane person can even do `obj.is_rhel_based = lambda: return False` :)

Reply:
> Some insane person can even do obj.is_rhel_based = lambda: return False :)

I will remember it =)
Anyway it's all about a matter of taste, so let's skip it.